core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -417,6 +417,13 @@ class ZkUtils(val zkClient: ZkClient,
     topicDirs.consumerOwnerDir + "/" + partition
   }
 
+  def controllerZkData(brokerId: Int, timestamp: Long): String = {
+    Json.encode(Map("version" -> 1, "brokerid" -> brokerId, "timestamp" -> timestamp.toString))
+  }
+
+  def preferredReplicaLeaderElectionZkData(partitions: scala.collection.Set[TopicAndPartition]): String = {
```

Comment:
These and other similar methods that don't access `ZkUtils` state should be in the companion object instead.